presto-main/src/main/java/com/facebook/presto/server/QueryProgressStats.java

```diff
@@ -0,0 +1,148 @@
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.joda.time.DateTime;
+
+import java.util.OptionalDouble;
+
+import static java.util.Objects.requireNonNull;
+
+public class QueryProgressStats
+{
+    private final DateTime executionStartTime;
+    private final long elapsedTimeMillis;
+    private final long queuedTimeMillis;
+    private final long cpuTimeMillis;
+    private final long scheduledTimeMillis;
+    private final long blockedTimeMillis;
+    private final long peakMemoryBytes;
+    private final long inputRows;
+    private final long inputBytes;
+    private final OptionalDouble progressPercentage;
+    private final boolean blocked;
+
+    @JsonCreator
+    public QueryProgressStats(
+            @JsonProperty("executionStartTime") DateTime executionStartTime,
+            @JsonProperty("elapsedTimeMillis") long elapsedTimeMillis,
+            @JsonProperty("queuedTimeMillis") long queuedTimeMillis,
+            @JsonProperty("cpuTimeMillis") long cpuTimeMillis,
+            @JsonProperty("scheduledTimeMillis") long scheduledTimeMillis,
+            @JsonProperty("blockedTimeMillis") long blockedTimeMillis,
+            @JsonProperty("peakMemoryBytes") long peakMemoryBytes,
+            @JsonProperty("inputRows") long inputRows,
+            @JsonProperty("inputBytes") long inputBytes,
+            @JsonProperty("blocked") boolean blocked,
+            @JsonProperty("progressPercentage") OptionalDouble progressPercentage)
+    {
+        this.executionStartTime = requireNonNull(executionStartTime, "executionStartTime is null");
+        this.elapsedTimeMillis = elapsedTimeMillis;
+        this.queuedTimeMillis = queuedTimeMillis;
+        this.cpuTimeMillis = cpuTimeMillis;
+        this.scheduledTimeMillis = scheduledTimeMillis;
+        this.blockedTimeMillis = blockedTimeMillis;
+        this.peakMemoryBytes = peakMemoryBytes;
+        this.inputRows = inputRows;
+        this.inputBytes = inputBytes;
+        this.blocked = blocked;
+        this.progressPercentage = requireNonNull(progressPercentage, "progressPercentage is null");
+    }
+
+    public static QueryProgressStats createQueryProgressStats(QueryStats queryStats)
+    {
+        return new QueryProgressStats(
+                queryStats.getExecutionStartTime(),
+                queryStats.getElapsedTime().toMillis(),
+                queryStats.getQueuedTime().toMillis(),
+                queryStats.getTotalCpuTime().toMillis(),
+                queryStats.getTotalScheduledTime().toMillis(),
+                queryStats.getTotalBlockedTime().toMillis(),
+                queryStats.getPeakMemoryReservation().toBytes(),
+                queryStats.getRawInputPositions(),
+                queryStats.getRawInputDataSize().toBytes(),
+                queryStats.isFullyBlocked(),
+                queryStats.getProgressPercentage());
+    }
+
+    @JsonProperty
```

Comment:
Please reorder all the getters to be in the same order as the fields (might be easiest to delete them and have IntelliJ generate them again, then add the annotation)